src/commands/aas/instrument.ts

```diff
@@ -98,14 +98,15 @@ https://docs.datadoghq.com/serverless/azure_app_services/azure_app_services_wind
     const siteContainers = await collectAsyncIterator(client.webApps.listSiteContainers(resourceGroup, aasName))
     const sidecarContainer = siteContainers.find((c) => c.name === SIDECAR_CONTAINER_NAME)
     const envVars = this.getEnvVars(config)
+    // We need to ensure that the sidecar container is configured correctly, which means checking the image, target port,
+    // and environment variables. The environment variables are must have the matching names and values as the
+    // keys in the App Settings, since the value indicates the name of the environment variable.
     if (
       sidecarContainer === undefined ||
       sidecarContainer.image !== SIDECAR_IMAGE ||
       sidecarContainer.targetPort !== SIDECAR_PORT ||
-      !equal(
-        Object.fromEntries(sidecarContainer.environmentVariables?.map(({name, value}) => [name, value]) ?? []),
-        envVars
-      )
+      !sidecarContainer.environmentVariables?.every(({name, value}) => name === value) ||
```

Comment:
Is it possible that `name !== value`? I feel like it's assumed to always be true since we create the sidecar, but in that case we would fallback in the `else` case, saying "Sidecar already exists with correct configuration".

I guess it could be clearer to fail early when `!sidecarContainer.environmentVariables?.every(({name, value}) => name === value)`, WDYT?